src/rust/engine/process_execution/src/remote.rs

```diff
@@ -587,7 +587,7 @@ impl CommandRunner {
         trace!("Got (nested) execute response: {:?}", execute_response);
         if execute_response.get_result().has_execution_metadata() {
           let metadata = execute_response.get_result().get_execution_metadata();
-          let parent_id = get_parent_id();
+          let parent_id = workunit_store::expect_workunit_state().parent_id;
```

Comment:
Can we always be sure that the workunit state will have been initialized when this is called?

Reply:
Yes, because there is only one path "into" the Graph, via https://github.com/pantsbuild/pants/pull/9522/files#diff-3c4ee36943d10de35244abe5aaf497fbR520.

There is a small amount of whack-a-mole here, in that as things are refactored in the `engine_cffi` crate (which we should continue to keep as small as possible), we'll need to move workunit initialization around. But having this assertion front and center like this in a commonly used codepath helps to ensure that a refactor won't break things in a way we won't notice.